pandas/core/arrays/string_.py

```diff
@@ -533,6 +533,11 @@ def _str_map_nan_semantics(
         else:
             return self._str_map_str_or_object(dtype, na_value, arr, f, mask)
 
+    def view(self, dtype: Dtype | None = None) -> ArrayLike:
+        if dtype is not None:
+            raise TypeError("Cannot change data-type for string array.")
+        return super().view(dtype=dtype)
```

Comment:
As an aside, I noticed NumPy's documentation on view says:

> Passing None for dtype is different from omitting the parameter, since the former invokes dtype(None) which is an alias for dtype('float64').

So I'm assuming we overload the meaning of `dtype=None` in our extension class hierarchy somewhere, otherwise this would fail (?)

Reply:
Not sure what you mean by "otherwise this would fail". I think we have behavior that differs from NumPy here:

https://github.com/pandas-dev/pandas/blob/fa5c2550e81c3e745eb7948b56adac45454853d5/pandas/core/arrays/base.py#L1874-L1876

Reply:
OK good to know we override. It wouldn't make sense otherwise to reinterpret these bytes using NumPy's default behavior (float)